pkg/controller/controller.go

```diff
@@ -811,12 +830,35 @@ func (h *Handler) generateDiskMakerDaemonSet(cr *localv1.LocalVolume) *appsv1.Da
 			},
 		},
 	}
+
+	err := addDaemonSetHash(ds)
+	if err != nil {
+		klog.Errorf("Unable to apply DaemonSet hash for diskmaker daemonset: %v", err)
+	}
+
 	h.applyNodeSelector(cr, ds)
 	addOwner(&ds.ObjectMeta, cr)
 	addOwnerLabels(&ds.ObjectMeta, cr)
 	return ds
 }
 
+func (h *Handler) checkDaemonSetHash(ds *appsv1.DaemonSet) bool {
+	forceRollout := false
+
+	existingDS, err := h.apiClient.getDaemonSet(ds.Namespace, ds.Name)
+	if err != nil {
```

Comment:
If there was an error fetching daemonset could it not be nil and taking annotations from a nil object will cause nil pointer dereference?

Reply:
I updated this to return if an error is detected. That should prevent any nil pointer dereferences.

If no existing DaemonSet can be found, then the sync* methods should continue with their previous behavior, as `daemonSetUpdated` will be false.